Add unused attribute to Dir_caches to avoid fatal errors with GCC 6
which treats all warnings as errors.
See https://sourceware.org/bugzilla/show_bug.cgi?format=multiple&id=19751
by Davide Vanzo (Vanderbilt University)

diff --git a/gold/arm.cc b/gold/arm.cc
index ed13c87..c47b002 100644
--- a/gold/arm.cc
+++ b/gold/arm.cc
@@ -597,7 +597,7 @@ class Reloc_stub : public Stub
 
     // Name of key.  This is mainly for debugging.
     std::string
-    name() const;
+    name() const ATTRIBUTE_UNUSED;
 
    private:
     // Stub type.
diff --git a/gold/dirsearch.cc b/gold/dirsearch.cc
index 1b3fefa..6178332 100644
--- a/gold/dirsearch.cc
+++ b/gold/dirsearch.cc
@@ -102,7 +102,7 @@ class Dir_caches
     : lock_(), caches_()
   { }
 
-  ~Dir_caches();
+  ~Dir_caches() ATTRIBUTE_UNUSED;
 
   // Add a cache for a directory.
   void add(const char*);